src/core/meshcop/border_agent.cpp

```diff
@@ -61,21 +61,46 @@ class ForwardContext
      *
      * @param[in]   aBorderAgent    A reference to the border agent.
      * @param[in]   aHeader         A reference to the request header.
-     * @param[in]   aSeparate       Whether this request should be responded separately.
+     * @param[in]   aPetition       Whether this request is leader petition.
```

Comment:
nit: Both LEAD_PET and COMM_PET messages exist.  Should we simplify to just say "petition"?
e.g. "is leader petition" -> "is a petition"